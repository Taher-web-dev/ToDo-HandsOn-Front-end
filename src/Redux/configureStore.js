import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { alertReducer } from './alert/alert';
import tasksReducer from './tasks/reducers/reducers';
import { itemReducer } from './item/item';

const reducer = combineReducers({
  alert: alertReducer,
  tasks: tasksReducer,
  item: itemReducer,
});
const store = createStore(reducer, applyMiddleware(thunk, logger));

export default store;
