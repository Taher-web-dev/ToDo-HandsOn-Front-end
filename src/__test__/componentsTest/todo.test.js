import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Todo from '../../components/todo/todo';

const mockStore = configureStore([]);
describe('Render correctly addItem component', () => {
  let store;
  let tree;
  beforeEach(() => {
    store = mockStore({
      alert: null,
      tasks: {},
      item: 'display',
    });
    store.dispatch = jest.fn();
    tree = render(
      <Provider store={store}>
        <Todo />
      </Provider>
    );
  });
  it('should render with given state from redux store', () => {
    expect(tree).toMatchSnapshot();
  });
});
