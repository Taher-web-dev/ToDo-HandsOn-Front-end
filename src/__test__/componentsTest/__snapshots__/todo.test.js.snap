// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly addItem component should render with given state from redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo-wrapper"
      >
        <header
          class="wrapper"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 date-day css-o2w69a-MuiTypography-root"
            style="font-size: 36px; font-weight: 700;"
          >
            Sunday, 24th
             
          </h1>
          <p
            class="date-month"
          >
            April
          </p>
          <div
            class="stat-wrapper"
          >
            <div
              class="stat-item"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 item-name css-5lbw0b-MuiTypography-root"
              >
                0
                 
              </h4>
              <p
                class="item-value"
              >
                Total
              </p>
            </div>
            <div
              class="stat-item"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 item-name css-5lbw0b-MuiTypography-root"
              >
                0
                 
              </h4>
              <p
                class="item-value"
              >
                Done
              </p>
            </div>
            <div
              class="stat-item"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 item-name css-5lbw0b-MuiTypography-root"
              >
                0
                 
              </h4>
              <p
                class="item-value"
              >
                Remaining
              </p>
            </div>
          </div>
        </header>
        <div
          class="functional-part"
          style="position: relative;"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 today-title css-5lbw0b-MuiTypography-root"
            style="font-size: 24px; font-weight: 550;"
          >
             Tasks for today 
          </h4>
          <button
            class="add-task-btn"
            style="position: absolute;"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            <p
              class="btn-title"
            >
              Add new item 
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo-wrapper"
    >
      <header
        class="wrapper"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 date-day css-o2w69a-MuiTypography-root"
          style="font-size: 36px; font-weight: 700;"
        >
          Sunday, 24th
           
        </h1>
        <p
          class="date-month"
        >
          April
        </p>
        <div
          class="stat-wrapper"
        >
          <div
            class="stat-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 item-name css-5lbw0b-MuiTypography-root"
            >
              0
               
            </h4>
            <p
              class="item-value"
            >
              Total
            </p>
          </div>
          <div
            class="stat-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 item-name css-5lbw0b-MuiTypography-root"
            >
              0
               
            </h4>
            <p
              class="item-value"
            >
              Done
            </p>
          </div>
          <div
            class="stat-item"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 item-name css-5lbw0b-MuiTypography-root"
            >
              0
               
            </h4>
            <p
              class="item-value"
            >
              Remaining
            </p>
          </div>
        </div>
      </header>
      <div
        class="functional-part"
        style="position: relative;"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 today-title css-5lbw0b-MuiTypography-root"
          style="font-size: 24px; font-weight: 550;"
        >
           Tasks for today 
        </h4>
        <button
          class="add-task-btn"
          style="position: absolute;"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            style="color: white;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <p
            class="btn-title"
          >
            Add new item 
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
